docs/V3/concepts/oracle.md

```diff
@@ -7,80 +7,94 @@ title: Oracle
+# Overview
+Oracles are smart contracts that communicate price feed information between assets traded on-chain. Their goal is to signal the trading price between a pair of assets and are often used as inputs to another contract’s execution. DeFi projects that move billions in value rely on price oracles to feed untampered data to their protocols.
 
-Historical data is stored as an array of observations. At first, each pool tracks only a single observation, overwriting it as blocks elapse. This limits how far into the past users may access data. However, any party willing to pay the transaction fees may increase the number of tracked observations (up to a maximum of `65535`), expanding the period of data availability to 9 days or more.
+Uniswap is an excellent data source for price oracles because of the sheer volume of trades. Exchanges with large volumes are likely to follow the “true” spot price of a trading pair and are more resistant to bad actors. (“True” refers to the hypothetical fair market value across all markets.)
 
-Storing price history directly in the pool contract substantially reduces the potential for logical errors on the part of the calling contract, and reduces integration costs by eliminating the need to store historical values. Additionally, the v3 oracle's considerable maximum length makes oracle price manipulation significantly more difficult, as the calling contract may cheaply construct a TWAP over an arbitrary position inside of, or encompassing, the full length of the oracle array.
+Uniswap v3 oracles return the average price and average liquidity (volume) of a traded pair within a given interval of seconds. All of this happens within a single call. The remainder of this page will explain how Uniswap oracles work at the protocol level and an exercise in using oracles.
 
+# How Uniswap’s Price Oracle Works
+Prices received from Uniswap’s oracles represent the average price of a trading pair over some time. Calculating the average price follows in a series of steps:
 
-## Observations
+1. At the beginning (or end) of each block, Uniswaps records price and time and aggregates them into a single cumulative value
+2. This value written to an array and stored in state
+3. The user calls a function and provides a time interval of n seconds from the current block timestamp
+4. The function returns an array with the cumulative value from the beginning and end of the time interval. From these numbers, users can calculate the average price of a trading pair.
 
-`Observation`s take the following form:
+## Observations, Ticks, and Arrays
+As with v2, Uniswap v3 uses an aggregate number that multiplies the two into an accumulated value. At the beginning of each block, Uniswap records the accumulated value of the first transaction that touches a token pool. Each observation contains:
```

Comment:
tickCumulative is not created by multiplying values, but rather adding the tick number to the previous value of tickCumulative